src/screens/profile/CreateAmenities/CreateAmenities.tsx

```diff
@@ -0,0 +1,193 @@
+
+type CreateAmenitiesNavigationProp = StackNavigationProp<RootStackParamList, 'CreateAmenities'>;
+type CreateAmenitiesRouteProp = RouteProp<RootStackParamList, 'CreateAmenities'>;
+
+export interface AmenitiesProps {
+  navigation: CreateAmenitiesNavigationProp;
+  route: CreateAmenitiesRouteProp;
+}
+
+const CreateAmenities = ({ route }: AmenitiesProps) => {
+  const { isNew, accomodationId } = route.params as { isNew: boolean; accomodationId: string };
+  const dispatch = useAppDispatch();
+  const navigation = useNavigation<NavigationProp<RootStackParamList>>();
+
+  const [selectedAmenities, setSelectedAmenities] =
+    useState<SelectedAmenities>(defaultAmenitiesState);
+  const [otherAmenities, setOtherAmenities] = useState<string[]>([]);
+  const [inputValue, setInputValue] = useState('');
+
+  const amenities: string[] = Object.keys(selectedAmenities);
+  const scrollViewRef = useRef<ScrollView>(null);
+
+  const scrollToBottom = () => {
+    if (scrollViewRef.current) {
+      scrollViewRef.current?.scrollToEnd({ animated: true });
+    }
+  };
+
+  const toggleAmenity = (amenityKey: string) => {
+    setSelectedAmenities((prevSelectedAmenities: SelectedAmenities) => ({
+      ...prevSelectedAmenities,
+      [amenityKey]: !prevSelectedAmenities[amenityKey],
+    }));
+  };
+
+  const addOtherAmenities = () => {
+    if (inputValue.trim() !== '' && inputValue.trim().length < 24) {
+      setOtherAmenities((prevAmenity) => [...prevAmenity, inputValue]);
+      scrollToBottom();
+      setInputValue('');
+    }
+  };
+
+  const renderAdditionalChips = () => {
+    return otherAmenities.map((item, index) => (
+      <Chip
+        key={index}
+        iconName={IconName.Check}
+        iconSet="ionicons"
+        text={item}
+        onTouchFunction={() => removeAdditionalChip(index)}
+        state
+      />
+    ));
+  };
+
+  const removeAdditionalChip = (index: number) => {
+    const updatedOptions = [...otherAmenities];
+    updatedOptions.splice(index, 1);
+    setOtherAmenities(updatedOptions);
+  };
+
+  const onFormSubmit = async () => {
+    try {
+      const response = await dispatch(
+        AsyncThunks.addAmenitiesThunk({
+          accomodationId,
+          data: {
+            ...selectedAmenities,
+            otherAmenities: otherAmenities.join(' ,'),
+          },
+        } as UpdateAmenitiesParams)
+      );
+      if (response.meta.requestStatus === 'fulfilled') {
+        console.log('Amenities added successfully');
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const getSavedAmenities = async () => {
+    const response = await dispatch(AsyncThunks.getAmenitiesThunk({ accomodationId }));
+    if (response.payload?.success) {
+      const { data } = response.payload as ApiSuccessResponseType<AccommodationAmenitiesResponse>;
+      setSelectedAmenities((({ otherAmenities, id, accomodationId, ...rest }) => rest)(data));
+      setOtherAmenities(data.otherAmenities.split(' ,'));
```

Comment:
does `data` field always have a `otherAmenities`?

Reply:
Yes. Backend always sends all possible amenities, including `otherAmenities` which can be empty string